### PR TITLE
Add new query stats metrics to track prometheus querystats

### PR DESCRIPTION
**Pull request by SungJin1212**
**What this PR does**:
Add new query stats metrics `cortex_query_samples_scanned_total` and `cortex_query_peak_samples`.
These track `scannedSamples` and `PeakSamples` for each in https://github.com/prometheus/prometheus/blob/main/util/stats/query_stats.go

**Which issue(s) this PR fixes**:
Fixes 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/frontend/transport/handler.go
Comment (yeya24) [inline pkg/frontend/transport/handler.go:134]: Can you please explain the usecase of tracking peak samples using a counter metric?

File: pkg/frontend/transport/handler.go
Comment (yeya24) [inline pkg/frontend/transport/handler.go:129]: I feel queryable samples is a bit confusing.
What about `cortex_query_samples_processed_total` or `cortex_query_samples_scanned_total`? 
You can add a comment to reference it to https://github.com/prometheus/prometheus/blob/main/util/stats/query_stats.go#L237

Comment (SungJin1212) [reply pkg/frontend/transport/handler.go:129]: For me, `cortex_query_samples_processed_total` is more good. Thank you!

File: pkg/frontend/transport/handler.go
Comment (SungJin1212) [reply pkg/frontend/transport/handler.go:134]: I think like promQL: `sum by (user) (irate(cortex_query_peak_samples_total[5m]))`, the cortex user can track the rate of how many samples are loaded in memory for each tenant's overall queries.

One usecase is: 
- The cortex user can track query memory usage (used by loaded samples) for each tenant and turn promQL to reduce query memory usage.

File: pkg/frontend/transport/handler.go
Comment (SungJin1212) [reply pkg/frontend/transport/handler.go:129]: How about changing the variable name `queryTotalQueryableSamples` to `queryProcessedSamples` and `querySamples` to `queryFetchedSamples`?

File: pkg/frontend/transport/handler.go
Comment (yeya24) [reply pkg/frontend/transport/handler.go:135]: OK. For a counter metric I feel it is more useful to just track total samples, not peak samples.
Would it be more useful to use native histogram instead?

Comment (SungJin1212) [reply pkg/frontend/transport/handler.go:135]: Yes, I think histogram is the best.

**Comment by SungJin1212** [comment]
@yeya24 
Could you please take a look?

File: pkg/frontend/transport/handler.go
Comment (yeya24) [reply pkg/frontend/transport/handler.go]: I think we can keep `queryTotalQueryableSamples` as it is clear that it refers to `Total Samples`.
Makes sense to rename `querySamples` to `queryFetchedSamples`.

**Review by yeya24 — APPROVED** [review]
Thanks. I think I am ok. Would be good to have another approval from maintainer before merging this.

**Comment by SungJin1212** [comment]
@alanprot 
could you please take a look? :)

**Comment by SungJin1212** [comment]
@alanprot @friedrichg
could you please take a look? :)

File: pkg/querier/stats/stats.proto
Comment (friedrichg) [inline pkg/querier/stats/stats.proto:44]: I am not sure I understand where the word "processed" comes from. Can you explain?

Maybe scanned_samples sounds better?

**Review by friedrichg — COMMENTED** [review]
Awesome tests. Small questions related to naming

File: CHANGELOG.md
Comment (SungJin1212) [reply CHANGELOG.md:11]: Thank you for catching it. `cortex_query_samples_processed_total` is right.

File: pkg/querier/stats/stats.proto
Comment (SungJin1212) [reply pkg/querier/stats/stats.proto:44]: `Processed` means the total number of considered samples when the query is executed.
But now, I feel like `scanned` is more intuitive to understand. Thanks.

File: pkg/frontend/transport/handler.go
Comment (SungJin1212) [reply pkg/frontend/transport/handler.go:95]: It looks good for consistency. Thanks you!

**Review by friedrichg — APPROVED** [review]

File: CHANGELOG.md
Comment (friedrichg) [inline CHANGELOG.md:12]:
```suggestion
* [ENHANCEMENT] Query Frontend: Add new query stats metrics `cortex_query_samples_scanned_total` and `cortex_query_peak_samples` to track scannedSamples and peakSample per user. #6228
```

**Review by yeya24 — APPROVED** [review]
Thanks!

**Pull request merged**
